core/src/impl/Kokkos_Profiling.cpp

```diff
@@ -332,7 +391,9 @@ void initialize() {
 #ifdef KOKKOS_ENABLE_LIBDL
   void* firstProfileLibrary = nullptr;
 
-  char* envProfileLibrary = getenv("KOKKOS_PROFILE_LIBRARY");
+  char* envProfileLibrary = (profileLibrary.empty())
+                                ? getenv("KOKKOS_PROFILE_LIBRARY")
+                                : const_cast<char*>(profileLibrary.c_str());
```

Comment:
This is now a bit outdated.
We should be doing
```C++
if (profileLibrary.empty())
  return;

char* envProfileLibrary = const_cast<char*>(profileLibrary.c_str());

Reply:
I expected you'd remove https://github.com/kokkos/kokkos/blob/a64a75f64d4113dfcf16a62bc12a9cb52b50fd19/core/src/impl/Kokkos_Profiling.cpp#L398-L402

Reply:
Ping